applications/tari_web_extension/src/lib.rs

```diff
@@ -0,0 +1,217 @@
+#[cfg(feature = "wee_alloc")]
+#[global_allocator]
+static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+
+lazy_static! {
+    static ref ICE_CNT: Arc<Mutex<u64>> = Arc::new(Mutex::new(0));
+    static ref SESSION: Arc<Mutex<String>> = Arc::new(Mutex::new("".to_string()));
+}
+
+async fn save_offer(session: &String, offer: &str) -> Result<(), JsValue> {
+    let url = format!("https://webrtc-test-d081d-default-rtdb.europe-west1.firebasedatabase.app/{session}/offer.json");
+    let body = format!("{{\"offer\": \"{}\"}}", offer.escape_default());
+
+    let mut opts = RequestInit::new();
+    opts.method("PUT");
+    opts.mode(web_sys::RequestMode::Cors);
+    // opts.headers(&JsValue::from_str("Content-Type: application/json"));
+    console_log!("{:?}", body);
+    opts.body(Some(&JsValue::from_str(body.as_str())));
+    let request = Request::new_with_str_and_init(&url, &opts)?;
+    let window = web_sys::window().ok_or("no global `window` exists")?;
+    JsFuture::from(window.fetch_with_request(&request)).await?;
+    Ok(())
+}
+
+fn save_ice_candidate(session: &String, ice: &RtcIceCandidate) -> Result<(), JsValue> {
+    let url;
+    {
+        let mut id = ICE_CNT.lock().unwrap();
+        url = format!(
+            "https://webrtc-test-d081d-default-rtdb.europe-west1.firebasedatabase.app/{session}/offer_ice/{}.json",
+            *id
+        );
+        *id += 1;
+    }
+
+    let mut opts = RequestInit::new();
+    opts.method("PUT");
+    opts.mode(web_sys::RequestMode::Cors);
+    // opts.headers(&JsValue::from_str("Content-Type: application/json"));
+    opts.body(Some(&JsValue::from_str(
+        format!("{}", JSON::stringify(&JsValue::from(ice.to_json())).unwrap()).as_str(),
+    )));
+    let request = Request::new_with_str_and_init(&url, &opts)?;
+    let window = web_sys::window().ok_or("no global `window` exists")?;
+    console_log!("saving ice");
+    #[allow(unused_must_use)]
+    {
+        // TODO: somehow solve the async call in sync closure, fortunately this is JsFuture and not rust. So it will
+        // run, we just don't wait for it.
+        JsFuture::from(window.fetch_with_request(&request));
+    }
+    Ok(())
+}
+
+#[wasm_bindgen(js_name = "getPc")]
+pub async fn get_pc() -> Result<JsValue, JsValue> {
+    let session = get_session_id().as_string().unwrap();
+    console_log!("Session id : {}", session);
+    let mut config = RtcConfiguration::new();
+    let ice_servers = Array::new();
+    let mut ice_server = RtcIceServer::new();
+    ice_server.url("stun:stun.l.google.com:19302");
+    ice_server.credential_type(web_sys::RtcIceCredentialType::Password);
+    ice_servers.push(&JsValue::from(ice_server));
+    config.ice_servers(&JsValue::from(&ice_servers));
+    let pc = RtcPeerConnection::new_with_configuration(&config)?;
+    Ok(JsValue::from(pc))
+}
+
+#[wasm_bindgen(js_name = "getDc")]
+pub async fn get_dc(pc: RtcPeerConnection, callback: JsValue) -> Result<JsValue, JsValue> {
```

Comment:
Could it be renamed to `get_data_channel`?

Reply:
done